dashboard/src/components/AppList/AppListGrid.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from "react";
+import { Link } from "react-router-dom";
+
+import { IAppOverview, IClusterServiceVersion, IResource } from "../../shared/types";
+import * as url from "../../shared/url";
+import { escapeRegExp } from "../../shared/utils";
+import CardGrid from "../Card/CardGrid.v2";
+import AppListItem from "./AppListItem.v2";
+
+import Alert from "../js/Alert";
+import "./AppList.v2.css";
+import CustomResourceListItem from "./CustomResourceListItem.v2";
+
+export interface IAppListProps {
+  appList: IAppOverview[] | undefined;
+  cluster: string;
+  namespace: string;
+  filter: string;
+  customResources: IResource[];
+  csvs: IClusterServiceVersion[];
+  appVersion: string;
+}
+
+function AppListGrid(props: IAppListProps) {
+  const { appList, customResources, cluster, namespace, appVersion, filter } = props;
+  const filteredReleases = (appList || []).filter(a =>
+    new RegExp(escapeRegExp(filter), "i").test(a.releaseName),
+  );
+  const filteredCRs = customResources.filter(cr =>
+    new RegExp(escapeRegExp(filter), "i").test(cr.metadata.name),
+  );
```

Comment:
Just for keeping components simpler, I use to move this kind of functions into a `component/utils/filters.tsx` folder.

Reply:
That's just hiding things under the carpet? I prefer not having the extra step if the filter is not going to be shared among different components.

Reply:
I always try to have components as simple as I can (in terms of code). The idea is to separate concerns and responsibilities. For components, they should:

1. Initialize based on properties
1. Run all the hooks to manage the state
1. Prepare stuff to be represented
1. JSX code

For this reason, I like to avoid having extra methods inside the components, but this is no a blocker :)